[Feature] Implements a scroll-to-top element for longer page contexts

egbadon-victor opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'.  -->
<!-- If there is no issue being resolved, please open one before creating this pull request. -->
<!-- Replace [issue number] with the issue number (don't leave the square brackets)--likewise for [issue author]. -->
- Fixes #54  by @possumbilities 

## Description
<!-- Concisely describe what the pull request does. -->
This PR implements a scroll-to-top element, which fixes the accessibility issue of going back to the top on a long page. 

## Technical details
<!-- Add any other information or technical details about the implementation; or delete this section entirely. -->
- Implements scroll-to-top class in the `vocabulary.css` file
- Implementation is demoed on the `archive-page.html` file in the `context` folder

## Tests
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or comment out this section entirely. -->
- Launch the `archive-page.html` file on your preferred browser and scroll
- Test that the scroll-to-top button is hidden until you scroll some distance
- Test that the scroll-to-top button is not visible on screens less than 700px
- Verify that there is no JS involved in this implementation

## Screenshots
<!-- Add screenshots to show the problem and the solution; or comment out this section entirely. -->

## Checklist
<!-- DON'T remove this section or any of the lines. -->
<!-- Leave incomplete or inapplicable lines unchecked. -->
<!-- Replace the [ ] with [x] to check the boxes (there is no space between x and square brackets). -->
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- YOU MUST READ AND UNDERSTAND THE FOLLOWING ATTESTATION. -->
<!-- -->
<!-- Be aware that copying and pasting from discussion sites or generative AI isn't allowed under this DCO. -->

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

brenoferreira commented on src/styles/form.scss:
```diff
@@ -6,7 +6,7 @@ $input-color: $color-dark-slate-gray;
 // text input
 
-.input {
+.input-text {
```
`.input` is a Bulma class https://bulma.io/documentation/form/input/. We just modify those base styles here a bit

brenoferreira commented on src/styles/typography.scss:
```diff
@@ -11,17 +11,17 @@ $font-sizes: (
 }
 
-.caption {
+.font-caption {
```
I think this is better as `.caption`. `.font-caption` looks a bit weird as class name.

hugosolar replied:
I'm going to resume this discussion. We should define a prefix for font-sizes and avoid class names who can lead to confusion. Also, we have to think these classes will be available to websites which will contain other class names and we should avoid future conflicts or trying to give some context for instance `.element .specific-class`

brenoferreira commented on src/styles/typography.scss:
```diff
@@ -30,24 +30,24 @@ $header-font-sizes: (
 }
 
-.value {
+.font-value {
```
Same here. I think this is better as `.value`. `.font-value` looks a bit weird as class name.

`.value` and `.caption` are the same as in the Figma designs

hugosolar replied:
Yes but it seems to be a little bit wide and general maybe we can give it some context.

brenoferreira replied:
`number-value` maybe? `caption` seems a bit obvious, but maybe `element-caption` is more clear. What do you think?

hugosolar replied:
same thing, I think it would be easier to standardize a prefix for font-sizes/properties this will be helpful to anyone who wants to work with the framework
, for instance, we can use the Bulma's naming convention for [font styles](https://bulma.io/documentation/modifiers/typography-helpers/#font-family)
But makes much more sense that way

egbadon-victor committed:
implements a scroll-to-top element

egbadon-victor committed:
Merge remote-tracking branch 'upstream/main' into scroll-to-top

possumbilities commented on src/css/vendor/normalize.css:
```diff
@@ -11,6 +11,7 @@
  html {
     line-height: 1.15; /* 1 */
     -webkit-text-size-adjust: 100%; /* 2 */
+    scroll-behavior: smooth;
```
As this is a vendored file, we can't make direct edits to it in this way. If this is a vital change to this PR it would need to occur elsewhere.

possumbilities commented on src/css/vocabulary.css:
```diff
@@ -1821,6 +1821,34 @@ body > footer .license svg {
 
+/* scroll to top element */
+
+.scroll-to-top{
```
missing space before `{`

possumbilities commented on src/css/vocabulary.css:
```diff
@@ -1821,6 +1821,34 @@ body > footer .license svg {
+    background-color: var(--vocabulary-brand-color-tomato);
+}
+
+.scroll-to-top::before{
```
missing space before `{`

possumbilities commented on src/css/vocabulary.css:
```diff
@@ -1821,6 +1821,34 @@ body > footer .license svg {
+}
+
+.scroll-to-top::before{
+    content:"↑";
```
we would want to include an appropriate icon (or use an existing icon) from within the `library-vars.css` file, to match visual styling.

possumbilities commented on src/css/vocabulary.css:
```diff
@@ -1821,6 +1821,34 @@ body > footer .license svg {
+    margin-top: calc(100vh + 200px);
+    margin-bottom: 10px;
+
+    background-color: var(--vocabulary-brand-color-dark-tomato);
```
a better background color might be the --vocabulary-neutral-lighter-grey as that would pair better with the current UX

egbadon-victor replied:
@possumbilities on implementing this, I realized that this is the same background color of the body of the page. That means that the background would be largely invisible. Hence, one would only be able to differenciate the scroll-to-top button by the icon alone.
Or perhaps you were referring to the color of the icon and not the background color

possumbilities commented:
Hi and thanks for this! 

In addition to the changes listed above, the button is flush to the right edge and should instead be aligned further into and possibly over the content area. 

The button also needs to be an element that could occur within the HTML of every page, but only become visible on longer pages. Please see the original Issue on the mention of pseudo classes to obscure the element.

> CSS pseudo elements using z-index stack positions and variable percentage based heights would offer a means to "reveal" the button after a degree of scrolling, and to visually hide the element on pages which do not reach a particular length.

egbadon-victor commented:
@possumbilities  thank you for taking your time to review this PR. I have noted all changes and concerns. I will get to work on implementing them Immedidately

egbadon-victor committed:
Merge branch 'creativecommons:main' into scroll-to-top

egbadon-victor committed:
makes improvements to scroll-to-top element and applies in all specimens

egbadon-victor committed:
Merge branch 'creativecommons:main' into scroll-to-top

egbadon-victor commented:
@possumbilities I have implemented all feedback given. 

I have also implemented it in such a way to it can be added to any page implementing vocabulary. It is hidden on shorter pages, and appears on longer pages without the use of JS.

I have also implemented it in all specimens, and updated the `docs/index.html` file to reflect and addition of a new component.

egbadon-victor committed:
Merge branch 'creativecommons:main' into scroll-to-top

egbadon-victor committed:
Merge branch 'main' into scroll-to-top

egbadon-victor committed:
Merge branch 'creativecommons:main' into scroll-to-top

egbadon-victor committed:
Merge branch 'creativecommons:main' into scroll-to-top

egbadon-victor committed:
Merge branch 'creativecommons:main' into scroll-to-top
